### ISSUE TITLE
Refactor FBetaScore

### ISSUE THREAD

**Issue by WindQAQ** [tensorflow/addons#2268]
**Describe the feature and the current behavior/state.**

Currently, [FBetaScore](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/f_scores.py) is not so maintainable and very different from how `tf.keras.metrics.*` handles similar things.

- `tf.keras.metrics.*` uses [metrics_utils.update_confusion_matrix_variables](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/utils/metrics_utils.py) to deal with all confusion matrix related metrics. It's a super helpful utility as it handles `sample_weight`, multi-label, multiple thresholds, e.t.c.. Unfortunately, it's not a public API.
- [tf.keras.metrics.AUC](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/metrics.py#L1847-L1930) handles multi-label with a argument `multi_label` passed into the constructor of the metric, but our `FBetaScore` handles it implicitly even without any shape check.

Given F-beta score is a function of precision and recall, I expect we can design either patterns in the following.

----------------

## Use `tf.keras.metrics.{Precision,Recall}`

```python
class FBetaScore(tf.keras.metrics.Metric):
    def __init__(self, *args, **kwargs):
        self.precision = tf.keras.metrics.Precision()
        self.recall = tf.keras.metrics.Recall()

    def update_state(self, y_true, y_pred, sample_weight=None):
        # Should also aggregate `sample_weight` to handle different weighting scheme
        return tf.group([
            self.precision.update_state(y_true, y_pred, sample_weight)
            self.recall.update_state(y_true, y_pred, sample_weight)
        ])

    def result(self):
        p = self.precision.result()
        r = self.recall.result()
        return some_interpolation_from_p_and_r
```

Pros:
- Clear as F-beta is the function of precision and recall.
- All public APIs.

Cons:
- `tf.keras.metrics.{Precision,Recall}` haven't supported multi-label data yet.
- true positives are computed twice in precision and recall.

## Use `metrics_utils.update_confusion_matrix_variables`

```python
class FBetaScore(tf.keras.metrics.Metric):
    def __init__(self, *args, **kwargs):
        # Follow how `tf.keras.metrics.AUC` constructs these variables
        self.true_positives = self.add_weight()
        self.false_positives = self.add_weight()
        self.false_negatives = self.add_weight()

    def update_state(self, y_true, y_pred, sample_weight=None):
        # Should also aggregate `sample_weight` to handle different weighting scheme
        return metrics_utils.update_confusion_matrix_variables(
          {
              metrics_utils.ConfusionMatrix.TRUE_POSITIVES:
                  self.true_positives,
              metrics_utils.ConfusionMatrix.FALSE_POSITIVES:
                  self.false_positives,
              metrics_utils.ConfusionMatrix.FALSE_NEGATIVES:
                  self.false_negatives,
          },
          y_true,
          y_pred,
          self._thresholds,
          sample_weight=sample_weight,
          multi_label=self.multi_label)

    def result(self):
        return some_interpolation_from_tp_fp_fn
```

Pros:
- Well support from `metrics_utils.update_confusion_matrix_variables`.
- Each variable is computed once.

Cons:
- Lots of copy-paste codes from `tf.keras.metrics.*`
- Not all APIs are public like `metrics_utils.update_confusion_matrix_variables`.

----------------

Both of them require support from core TF, but I prefer option two. Use of that private API also simplies confusion matrix related metrics in addons such as `MultiLabelConfusionMatrix`, and enables extra functionality like multiple thresholds.

cc @seanpmorgan and @bhack. Any thought on this? Thank you 😃

**Comment by AakashKumarNain** [comment]
The pros of second option make it more preferable

**Comment by bhack** [comment]
From the last Keras meeting, as Keras Will go standalone again quite soon reopeing Its own repository I suppose It will be easier to Request some public API switch for the solution  2).

**Comment by chazuttu** [comment]
Using metrics_utils.update_confusion_matrix_variables would be better.

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
